bindings/native/src/wrapper.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//#![allow(unused_imports)]
+use crypto::keys::slip10::ChainCode;
+use iota_stronghold_new as stronghold;
+use iota_stronghold_new::procedures::WriteVault;
+use stronghold::{
+    procedures::{Chain, Ed25519Sign, GenerateKey, KeyType, PublicKey, Slip10Derive, Slip10Generate},
+    Client, KeyProvider, Location, SnapshotPath, Stronghold,
+};
+use thiserror::Error as DeriveError;
+
+const CLIENT_PATH: &str = "wasp";
+const VAULT_PATH: &str = "wasp";
+const KEY_TYPE: KeyType = KeyType::Ed25519;
+const SEED_LENGTH: usize = 32;
+const RECORD_PATH_SEED: &str = "seed";
+
+pub struct StrongholdWrapper {
+    snapshot_path: String,
+    stronghold: Stronghold,
+    client: Client,
+}
+
+#[derive(Debug, DeriveError)]
+#[non_exhaustive]
+pub enum WrapperError {
+    #[error("Failed to open snapshot")]
+    OpenSnapshot,
+
+    #[error("Failed to commit to snapshot")]
+    CommitToSnapshot,
+
+    #[error("Failed to create client")]
+    CreateClient,
+
+    #[error("Failed to write client")]
+    WriteClient,
+
+    #[error("Failed to execute procedure")]
+    ExecuteProcedure,
+}
+
+impl StrongholdWrapper {
+    pub fn from_file(snapshot_path: String, key_as_hash: Vec<u8>) -> Result<Self, WrapperError> {
```

Comment:
```suggestion
    pub fn from_file<R>(snapshot_path: String, key_as_hash: R) -> Result<Self, WrapperError> where R : AsRef<[u8]> {
```

You can use a bounded trait for this.